lib/avromatic/io/datum_reader.rb

```diff
@@ -1,86 +1,37 @@
 # frozen_string_literal: true
 
-# rubocop:disable Style/WhenThen
 module Avromatic
   module IO
     # Subclass DatumReader to include additional information about the union
     # member index used. The code modified below is based on salsify/avro,
     # branch 'salsify-master' with the tag 'v1.9.0.3'
     class DatumReader < Avro::IO::DatumReader
 
-      UNION_MEMBER_INDEX = Avromatic::IO::UNION_MEMBER_INDEX
-
-      def read_data(writers_schema, readers_schema, decoder, initial_record = nil)
-        # schema matching
-        unless self.class.match_schemas(writers_schema, readers_schema)
-          raise Avro::IO::SchemaMatchException.new(writers_schema, readers_schema)
-        end
-
+      def read_data(writers_schema, readers_schema, decoder)
         # schema resolution: reader's schema is a union, writer's schema is not
-        if writers_schema.type_sym != :union && readers_schema.type_sym == :union
-          rs_index = readers_schema.schemas.find_index do |s|
-            self.class.match_schemas(writers_schema, s)
-          end
+        return super unless writers_schema.type_sym != :union && readers_schema.type_sym == :union
```

Comment:
For my own understanding, this is the change that references reducing the maintenance burden right?

Reply:
Yup. This change + removing the override of `read_record` (which was missing https://github.com/apache/avro/pull/636 from upstream).